subprojects/core/src/main/java/org/gradle/process/internal/JvmOptions.java

```diff
@@ -135,8 +140,13 @@ protected void formatSystemProperties(Map<String, ?> properties, List<String> ar
         if (assertionsEnabled) {
             args.add("-ea");
         }
+
         if (debug) {
-            args.add("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005");
+            if (debugPort <= 0) {
```

Comment:
❌ We shouldn't bind the debug port to whether or not server or suspend are used

Reply:
The idea here was to keep the existing `--debug-jvm` functionality.

Reply:
I've refactored the code.